### ISSUE TITLE
[Help]Cannot run 'node' or 'npm' command after nvm installed

### ISSUE THREAD

**Issue by yuanhongb** [coreybutler/nvm-windows#221]
I'm trying to install nvm on my windows8.1(64bits). I do as the below orders:
1. Download zip file.
2. Unzip and install it.
3. Configure the env path.
4. Try to run 'nvm v' and it works well.
![image](https://cloud.githubusercontent.com/assets/9713291/20034911/3786820a-a40b-11e6-95ab-372e20dc1a93.png)

5. Try to install nodejs ```nvm install 4.2.4 64```, after a while it tells me installation completed and suggests to run ```nvm use 4.2.4``` to use it.
![image](https://cloud.githubusercontent.com/assets/9713291/20034936/181a4144-a40c-11e6-927d-57db48791203.png)
6. I run ```node -v``` to see if it really works. Unfortunately, I met some error msg.
![image](https://cloud.githubusercontent.com/assets/9713291/20034979/2caca614-a40d-11e6-8d7c-5142ab24d09f.png)
It means 'This application cannot be run on your computer.'

It confused me. Did anyone meet the same error before? And How to solve it ??

**Comment by espoelstra** [comment]
You will need to run this inside an "Administrator" or elevated command prompt. To change/create the symlinks nvm requires Administrator permissions.

**Comment by summerinsect** [comment]
I encountered the same problem

**Comment by yuanhongb** [comment]
@summerinsect Do u fix it?

**Comment by andrewbhy** [comment]
I have the same problem.  Currently using Windows 10 and have Windows Defender running.  

1. Windows Defender doesn't recognize any threats while running the nvm install.
2. I am using administrator mode for the command prompt
3. I  do not see any error message during the node install : 

![nvm_problem_01](https://cloud.githubusercontent.com/assets/12205434/20541668/0fe40ccc-b0b3-11e6-94f1-ae36ae9a3cbb.PNG)

4. nvm list gives me the following output :
![nvm_problem_02](https://cloud.githubusercontent.com/assets/12205434/20541439/1352cce6-b0b2-11e6-8346-ec4dd4b8e012.PNG)

5. c:\program files\nodejs folder only contains node_modules folder which makes me think that the nvm installation wasn't completed properly.


Any suggestions?

**Comment by yuanhongb** [comment]
@andrewbhy Cool, it works! Thank you so much!

**Comment by yuanhongb** [comment]
Unfortunately, the issue haven't been fixed yet. :( Now, I could run node command, but couldn't run npm command:
![image](https://cloud.githubusercontent.com/assets/9713291/20640395/beb4f790-b417-11e6-80d0-c70c1b0b57c5.png)

Is there anything I missed??

**Comment by bmcminn** [comment]
Steps provided above didn't work for me... I have `NVM_HOME` and `NVM_SYMLINK` configured in my local and system path as variables, and included them in my system `PATH`, however, I run `nvm use *.*.*` and no matter which version I switch to, NVM creates the symlink, but doesn't persist the Node instance I wish to use.

Far as I can tell, everything is where it needs to be, but NVM isn't persisting which Node version I've set using `nvm use`.

**Comment by dragon788** [comment]
I don't think you want to have the NVM_HOME and NVM_SYMLINK in both places. What do you have in your settings.txt file?

**Comment by bmcminn** [comment]
@dragon788 I figured out the issue, but it wasn't related to the env path variables. Mind you, NVMs installer setup both variables in both places, so that was not a decision made on my part.

The fix for my issue is detailed in #230

**Comment by fresheneesz** [comment]
@coreybutler I have the same problem. After installation, I can run it in admin mode, but not in normal user mode. I seem to remember last time I used this (months ago at this point), that I could use nvm from whereever, but if an installation needed to happen, a prompt would come up. That's a much better experience - you shouldn't have to launch an admin cmd prompt

**Comment by coreybutler** [comment]
If you're downloading the manual installer, this is the experience you can likely expect because nothing is setup for you. This is why I made the installer. It prepares everything for you, including permissions. Keep in mind it is not enough to run this as an administrator on some versions of Windows. Later versions of Windows requires _elevated_ admin privileges, though in most scenarios it should prompt you when you install a new version or make a switch.

If you install NVM4W anywhere within `C:\Program Files\`, you **must** use an account with admin privileges. This is a restriction of Windows itself, which considers this a protected directory.

If you're having problems with persisting a version of Node, make sure your account has write permissions on the NVM for Windows user data directory. This is required to write the `settings.txt` file.

If you're on Windows 7, please read about [Windows 7 support](https://github.com/coreybutler/nvm-windows/wiki/Support-for-Windows-7-and-Earlier).

**Comment by fresheneesz** [comment]
@coreybutler I'm not sure what you mean by "manual installer" vs "the installer". I used https://github.com/coreybutler/nvm-windows/releases/download/1.1.1/nvm-setup.zip . My account has admin privileges, but its annoying to have to manually start a cmd prompt with admin privileges in order to access it. 

Are you saying if I install nvm somewhere other than program files it'll work without elevated privileges?

**Comment by coreybutler** [comment]
@fresheneesz - sorry for being as clear as mud :) By "manual" installer, I meant the "nvm-noinstall.zip". You're using "the installer", i.e. the GUI that walks you through the setup process.

Depending on your version of Windows, installing elsewhere only requires admin privileges, not elevated admin privileges. It still needs admin privileges whenever it switches to a new version though (symlinking). 

@fresheneesz - I just ran a fresh install in Windows 10 and it prompts when switching versions, so I'm not sure why you're having to open a new prompt with admin privileges. Do you have something like cmder/cygwin/git shell installed? Sometimes these shells cause problems like this.

**Comment by fresheneesz** [comment]
I seem to have gotten it to work, but I think my computer might be always opening up a cmd prompt with admin mode now? I'm not sure, it suddenly switched to always being green on black, which I thought was admin mode. IDK but it works now so yay!

**Comment by dragon788** [comment]
Green on Black is the "cmd" prompt. White on Blue is the "PowerShell" prompt. You can usually tell the Administrator permissions in the title bar, it will say "Administrator" up there and sometimes will start in `C:\Windows\system32` when it first opens.

**Comment by helzgate** [comment]
Your going to think I'm crazy but I couldn't npm to work because the files weren't being created for npm deep in the directory tree within c:\Program Files\nodejs\node_mudules\npm  No matter how I ran the cmd.exe wether as administrator or not.  For giggles I ran the powershell as admin and it had no problem creating the files on Windows Server 2012 R2.  What powershell does differently than cmd.exe I have no idea.  One annoying thing though, it's slow.

**Comment by helzgate** [comment]
..Er, well, all kinds of craziness.  Had issues with v8.4.0 even though I could set/use 6.11.3 with no problem.  I ended up uninstalling/removing v8.4.0 and this time when I reinstalled v8.4.0 I left off the "v" and it worked.

**Comment by dragon788** [comment]
@helzgate cmd.exe is limited to the 256 character path depth I believe while the PowerShell "shell" doesn't seem to suffer from that limitation. Weird that the `v` vs no `v` makes a difference, I'd hope they would use mostly the same logic and just strip the preceding character but I haven't checked out that code in depth.

**Comment by githubbob42** [comment]
I had the same problem (I could only run `nvm` and `node` as the admin after default install).  However, if you choose different install directories, other than the defaults (I used `c:/nodejs` for Node and `c:/bin/nvm` for NVM) it works perfectly.  I no longer need to be admin to run `node` or `nvm`.  No special setup or configuration, just select different install locations and that was it.

**Comment by dragon788** [comment]
When I put together the Chocolatey package I believe I had it install to C:\ProgramData because that is accessible by all users but only writable by administrators, but that means as long as you run your `nvm install` and `npm install -g` type commands from an administrator prompt everything else should work without the space issue I think.

**Comment by SanjanaTailor** [comment]
using nvm windows how to install cordova ?

**Comment by psiddharth709** [comment]
![error](https://user-images.githubusercontent.com/25383528/30782518-5a8d99ca-a151-11e7-801b-ce637b4eea8c.png)


how to solve error?? please any idea share 

windows 7 32 bit not install nvm using cmd ..

**Comment by psiddharth709** [comment]
npm install -g cordova

**Comment by SanjanaTailor** [comment]
Facing this problem after installing nvm 
module.js:471
    throw err;
    ^

Error: Cannot find module 'C:\Windows\System32\node_modules\cordova\bin\cordova'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:383:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:496:3

E:\Cordova projects\CordovaApp>nvm ls

  * 6.11.4 (Currently using 64-bit executable)
Facing this problem can not run cordova for 2 months. Help me

**Comment by coreybutler** [comment]
@SanjanaTailor - did you have a prior version of Node installed before using nvm4w? See https://github.com/coreybutler/nvm-windows#installation--upgrades

**Comment by ChuckkNorris** [comment]
If anyone else encountered an issue where the `node` or `npm` commands were not found after installing and using a version, I managed to resolve this by turning `nvm` on and off:

`nvm off`

`nvm on`

![example](https://i.imgur.com/6og3Hio.png)

**Comment by noodlesegg** [comment]
Excellent @ChuckkNorris

**Comment by dvnrsn** [comment]
I tried @ChuckkNorris trick without success. What appears to have worked for me was after `nvm install latest` including in Path:

`C:\Users\<user>\AppData\Roaming\nvm\v9.8.0`

**Comment by csempesz** [comment]
This is a nightmare. I mean to make this work. I tried all the above: install nvm into something else then Program Files, used PowelShell (Admin) to launch the install, switched on/off nvm, basically none of them worked. 
Strange thing is that when I tried to install into something else then Program Files (like c:\nodejs), nvm install deleted the target directory and nothing got installed.
I'm using Windows 10. At this poitn I'm not recommending using this product to manage multiple node version.

**Comment by b-dur** [comment]
Hi @csempesz. Could you try uninstall NVM and then install this pre-release and see if it solves your problem?
https://github.com/b-dur/nvm-windows/releases/tag/1.1.7

**Comment by emekaokoli** [comment]
I fixed mine like this...
nvm on,
And then installed a single node version as admin with powershell. 
I am using windows 10.

**Comment by gdutleung** [comment]
Just use use powershell rather than cmd and try again.

**Comment by martyglaubitz** [comment]
@csempesz dont install it via the installer - use chocolatey. This in conjunction with turning nvm on and off, solved it for me

**Comment by adamorlowskipoland** [comment]
> 3\. nvm on

Fixed all bugs for me.

**Comment by mynormh** [comment]
Still getting this problem with 1.1.7 on windows 10. node -v works fine but npm -v gives me:
![image](https://user-images.githubusercontent.com/2418695/50404423-5c20e480-076d-11e9-8186-a197457d6f6a.png)

**Comment by papb** [comment]
I finally fixed this in Windows 7 by:

* Uninstalling nvm
* Installing it on nondefault folders (`C:\nvm` and `C:\nodejs`)
* Altering manually the `%PATH%` to use these paths directly instead of `%NVM_HOME%` and `%NVM_SYMLINK%`
* `nvm install 10.15.0`
* `nvm off`
* `nvm on`
* `nvm use 10.15.0`

**Comment by ilog2000** [comment]
Just run into this issue with nvm v1.1.7 on Windows 10. The solution in my case was to completely delete 'C:\Program Files\nodejs' directory (it existed after NodeJS uninstall). After that ```nvm use 11.8.0``` started working normally.

**Comment by mynormh** [comment]
Thanks! I’ll try to do this and see if it works.

**Comment by alhadathavale** [comment]
> Excellent @ChuckkNorris



> If anyone else encountered an issue where the `node` or `npm` commands were not found after installing and using a version, I managed to resolve this by turning `nvm` on and off:
> 
> `nvm off`
> 
> `nvm on`
> 
> ![example](https://camo.githubusercontent.com/58f4607c3856ed3e5ee08908565a6553d8b3481c/68747470733a2f2f692e696d6775722e636f6d2f366f673348696f2e706e67)

That works!

**Comment by alhadathavale** [comment]
@ChuckkNorris That worked! Thx!

**Comment by salvimateus** [comment]
> Basta encontrar esse problema com o nvm v1.1.7 no Windows 10. A solução no meu caso foi excluir completamente o diretório 'C: \ Arquivos de Programas \ nodejs' (ele existia após a desinstalação do NodeJS). Depois disso `nvm use 11.8.0`começou a funcionar normalmente.

PERFECT!

**Comment by salvimateus** [comment]
@ilog2000 PERFECT! Thank you

**Comment by presiden** [comment]
> If anyone else encountered an issue where the `node` or `npm` commands were not found after installing and using a version, I managed to resolve this by turning `nvm` on and off:
> 
> `nvm off`
> 
> `nvm on`
> 
> ![example](https://camo.githubusercontent.com/7921366e66ff2cf4abfa8208d70a5625cb410ba27576834e38ee02fd38d17963/68747470733a2f2f692e696d6775722e636f6d2f366f673348696f2e706e67)

Great! it works. Just run as admin

**Comment by LeedsCode** [comment]
**tldr;** If none of the methods before work, the specific version of node you're trying to use may have installed incorrectly. Try uninstalling it and reinstalling using `nvm uninstall {version}` and `nvm install {version}`

All of the previous methods did not work for me. What went wrong for me is one of the versions of Node I installed using `nvm install` did not install correctly. I actually switched to a VPN while it was downloading. It showed an error, but it still made it seem like it worked as I could use it. However it gave me the same error. I switched to a different version of node and that one was working, but it wouldn't work with this latest version I had installed. After I uninstalled that version and reinstalled it, it's working again!

**Comment by phaddmin** [comment]
> I finally fixed this in Windows 7 by:
> 
> * Uninstalling nvm
> * Installing it on nondefault folders (`C:\nvm` and `C:\nodejs`)
> * Altering manually the `%PATH%` to use these paths directly instead of `%NVM_HOME%` and `%NVM_SYMLINK%`
> * `nvm install 10.15.0`
> * `nvm off`
> * `nvm on`
> * `nvm use 10.15.0`

Using powershell as admin every time was not an option for me... With NVM v1.1.11, the above worked for me but only needed to do the first 2 steps; uninstall nvm, install again on non-default folders directly under C: drive. I didnt need to update the NVM_HOME and NVM_SYMLINK variables.
